oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/multiplex/MultiplexingNodeStore.java

```diff
@@ -99,11 +99,11 @@ public NodeState merge(NodeBuilder builder, CommitHook commitHook, CommitInfo in
         processed.compareAgainstBaseState(builder.getNodeState(), new ApplyDiff(nodeBuilder));
 
         Map<MountedNodeStore, NodeState> resultStates = newHashMap();
-        for (Map.Entry<MountedNodeStore, NodeBuilder> e : nodeBuilder.getBuilders().entrySet() ) {
-            NodeStore nodeStore = e.getKey().getNodeStore();
-            NodeBuilder rootBuilder = e.getValue();
-            NodeState result = nodeStore.merge(rootBuilder, EmptyHook.INSTANCE, info);
-            resultStates.put(e.getKey(), result);
+        for (MountedNodeStore mountedNodeStore : ctx.getAllMountedNodeStores()) {
```

Comment:
I thought that the constant ordering may be useful in the future, for the concurrency support (but obviously it's not enough).